src/org/rascalmpl/values/RascalFunctionValueFactory.java

```diff
@@ -105,7 +105,7 @@ private ParserGenerator getParserGenerator() {
 
     private Class<IGTD<IConstructor, ITree, ISourceLocation>> generateParser(IMap grammar) {
         try {
-            return getParserGenerator().getNewParser(new NullRascalMonitor(), URIUtil.rootLocation("parser-generator"), "$GENERATED_PARSER$" + Math.abs(grammar.hashCode()), grammar);
+            return getParserGenerator().getNewParser(ctx.getEvaluator().getMonitor(), URIUtil.rootLocation("parser-generator"), "$GENERATED_PARSER$" + Math.abs(grammar.hashCode()), grammar);
```

Comment:
this can return `null` is that a problem?

Reply:
For robustness sake, the Evaluator checks for null monitors always when the monitor API is being used, at every call; so no this is ok. Still a sharp observation!